include/stdexec/__detail/__intrusive_queue.hpp

```diff
@@ -87,7 +87,7 @@ namespace stdexec {
         void push_back(_Item* __item) noexcept {
           STDEXEC_ASSERT(__item != nullptr);
           __item->*_Next = nullptr;
-          if (__tail_ == nullptr) {
+          if (empty()) {
```

Comment:
`empty()` checks whether `__head_` is `nullptr`. I haven't spotted the bug yet. It looks like when `__head_` is null, `__tail_` is also null. How do we get into a state where only one is null?

Reply:
I traced the issue to the thread enqueueing a task that was never consumed. I intuited the behavior of `push_back()` only setting `__head_` when `__tail_` is null could be a bug, especially since `__tail_` is set unconditionally at the bottom.

But as we've discussed offline, I never suspected the real culprit could be an `nvc++` bug in the `std::exchange()` call in `pop_front()`.